03week/towersOfHanoi.js

```diff
@@ -19,24 +19,44 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(start, end) {
   // Your code here
-
+  console.log("made it to MovePiece");
+  var disk = start.pop();
+  end.push(disk);
+  return checkForWin(end);
+  //return stacks;
 }
+//console.log(stacks);
+//console.log(movePiece('a', 'c'));
+//console.log(stacks);
 
-function isLegal() {
+function isLegal(start, end) {
   // Your code here
-
+  //if it's not legal, say "not a legal move"
+  //if (end[end.length-1])
+  if (end[end.length-1] < start[start.length-1]){
+    console.log("illegal move");
+  }
+  else {
+    return movePiece(start, end);
+  }
+  //if legal eturn movePiece(startStack, endStack);
 }
 
-function checkForWin() {
+function checkForWin(end) {
   // Your code here
-
+  console.log("made it to check for win");
+  if(end === [4, 3, 2, 1]) {
```

Comment:
you're close. think about the winning conditions for the game, like I mentioned in class last night. For a valid win, we're really only concerned with stacks *other than* the starting stack (e.g. `stacks.b` or `stacks.c`). If the `isLegal` restrictions are in place, we can safely assume that the pieces being moved to `stacks.b` or `stacks.c` are supposed to be there.

Since the stacks are Arrays, I'm thinking `.length` would be a better way to go about solving this problem.

```javascript
function checkForWin() {
  if (/* stacks.b or stacks.c have a length of 4 */) {
    console.log("You win!");
  }
}
```
 Notice we don't need to pass anything to the function - since `stacks` is a globally-available variable, we can reference it anywhere in the program.